### PR TITLE
Make `--engine-debug` easy to use

### PR DESCRIPTION
**Pull request by W95Psp**
This PR:
 - rewrites the tiny webserver serving the webapp for debugging the engine in Rust;
 - embed this webserver directly into the Rustc driver;
 - when `--engine-debug` (or `-d`) is passed to the CLI, the driver now spawns a webserver directly so that debugging is very easy.

(this is somehow related to #121)

### CODE REVIEW

File: engine/utils/phase_debug_webapp/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+use phase_debug_webapp::*;
+
+fn main() {
+    run(|| std::fs::read_to_string("/tmp/hax_debug/debug-hax-engine.json").unwrap())
```
Comment (franziskuskiefer) [inline engine/utils/phase_debug_webapp/src/main.rs:4]: A hard-coded path? I'm not sure that's a good idea....

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Yay, only one comment

File: engine/utils/phase_debug_webapp/src/main.rs (hunk 1 above)
Comment (W95Psp) [reply engine/utils/phase_debug_webapp/src/main.rs:4]: Oops, that's was not intended

Comment (W95Psp) [reply engine/utils/phase_debug_webapp/src/main.rs]: Fixed it: I actually removed this binary entirely, it has no purpose now

File: engine/DEV.md [hunk 2]
```diff
@@ -46,10 +46,7 @@ You can also use `dune utop` and then `#show_type Hax_engine.Types.SOME_TYPE` an
...
-Just add the flag `--debug-engine DIR` to the `into` subcommand, where
-`DIR` is a directory where will be outputted debug informations. Make
-sure to create the folder beforehand.
-
-To browse the debug information, run
-`./engine/utils/phase_debug_webapp/server.js DIR`. Note you can change
-to port by setting the environment variable `PORT`.
+Just add the flag `--debug-engine` (or `-d`) to the `into` subcommand.
+At the end of the translation, `cargo hax` will spawn a webserver that
+let you browse the debug information. Note you can change to port by
```
Comment (franziskuskiefer) [inline engine/DEV.md:51]:
```suggestion
lets you browse the debug information. Note, you can change to port by
```

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**